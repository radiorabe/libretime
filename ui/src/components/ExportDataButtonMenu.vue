<template>
  <v-menu offset-y>
      <template v-slot:activator="{ on, attrs }">
        <v-btn
          color="primary"
          dark
          v-bind="attrs"
          v-on="on"
        >
          {{ $t('Exportâ€¦') }}
        </v-btn>
      </template>
      <v-list>
        <v-list-item>
          <v-list-item-title>CSV</v-list-item-title>
        </v-list-item>
        <v-list-item>
          <v-list-item-title>JSON</v-list-item-title>
        </v-list-item>
      </v-list>
  </v-menu>
</template>

<script>
  export default {
    name: 'ExportDataButtonMenu',
    props: {
      data: String,
    }
  }
</script>
