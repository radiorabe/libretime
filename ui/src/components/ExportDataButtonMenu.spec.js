import { createLocalVue, mount } from '@vue/test-utils'

import Vue from 'vue'
import Vuetify from 'vuetify'

import ExportDataButtonMenu from '../../src/components/ExportDataButtonMenu.vue';

describe('ExportDataButtonMenu.vue', () => {
  const localVue = createLocalVue()
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should match snapshot', () => {
    const wrapper = mount(ExportDataButtonMenu, {
      localVue,
      vuetify,
      mocks: {
        // Mock VueI18n
        $t: () => {}
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
