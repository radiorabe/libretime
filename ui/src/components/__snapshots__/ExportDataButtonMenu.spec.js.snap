// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportDataButtonMenu.vue should match snapshot 1`] = `
<div class="v-menu"><button type="button" class="v-btn v-btn--contained theme--dark v-size--default primary" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content">

      </span></button>
  <!---->
</div>
`;
